metis-enrichment/metis-enrichment-client/src/main/java/eu/europeana/enrichment/rest/client/enrichment/EnricherImpl.java

```diff
@@ -0,0 +1,109 @@
+package eu.europeana.enrichment.rest.client.enrichment;
+
+import static eu.europeana.metis.utils.ExternalRequestUtil.retryableExternalRequestForNetworkExceptions;
+
+import eu.europeana.corelib.definitions.jibx.RDF;
+import eu.europeana.enrichment.api.external.model.EnrichmentBase;
+import eu.europeana.enrichment.api.external.model.EnrichmentBaseWrapper;
+import eu.europeana.enrichment.api.external.model.EnrichmentResultList;
+import eu.europeana.enrichment.rest.client.exceptions.EnrichmentException;
+import eu.europeana.enrichment.utils.EnrichmentFields;
+import eu.europeana.enrichment.utils.EnrichmentUtils;
+import eu.europeana.enrichment.utils.EntityMergeEngine;
+import eu.europeana.enrichment.utils.InputValue;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EnricherImpl implements Enricher{
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Enricher.class);
+
+  private final EntityMergeEngine entityMergeEngine;
+  private final EnrichmentClient enrichmentClient;
+
+  public EnricherImpl(EntityMergeEngine entityMergeEngine,
```

Comment:
This contradicts the Builder pattern where the user can actually create this object by using this constructor(normally the constructor should be private)..
Are we allowing for that?
If yes then perhaps the Builder naming should change to something else to avoid confusion..

(This looks more like a factory pattern to me)

Reply:
I changed the name of the class just like we discussed